streams-bootstrap-test/src/main/java/com/bakdata/kafka/KafkaTestClient.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2025 bakdata
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.bakdata.kafka;
+
+import static java.util.Collections.emptyMap;
+
+import com.bakdata.kafka.util.ImprovedAdminClient;
+import com.bakdata.kafka.util.TopicClient;
+import com.bakdata.kafka.util.TopicSettings;
+import com.bakdata.kafka.util.TopicSettings.TopicSettingsBuilder;
+import java.util.Map;
+import lombok.NonNull;
+import lombok.RequiredArgsConstructor;
+
+/**
+ * Client that supports communication with Kafka clusters in test setups, including topic management, reading from
+ * and sending to topics.
+ */
+@RequiredArgsConstructor
+public class KafkaTestClient {
+
+    private static final TopicSettingsBuilder DEFAULT_TOPIC_SETTINGS = TopicSettings.builder()
+            .partitions(1)
+            .replicationFactor((short) 1);
+    private final @NonNull KafkaEndpointConfig endpointConfig;
+
+    /**
+     * Create q new {@code TopicSettingsBuilder} which uses a single partition and no replicas
+     * @return default topic settings
+     */
+    public static TopicSettingsBuilder defaultTopicSettings() {
+        return DEFAULT_TOPIC_SETTINGS;
```

Comment:
Isn't lombok's builder mutable? So if you use this in multiple places, they share one instance and therefore also see the changes of each other?